app/services/subscriptions/dates_service.rb

```diff
@@ -0,0 +1,304 @@
+# frozen_string_literal: true
+
+module Subscriptions
+  class DatesService < BaseService
+    def initialize(subscription, timestamp)
+      @subscription = subscription
+
+      # NOTE: timestamp is the billing day
+      #       It should be the end of the billing period + 1 day
+      #       When subscription is terminated, it is the termination day
+      @timestamp = timestamp
```

Comment:
Is it not simpler to directly send and use the `billing_date` instead of a `timestamp`? 🤔